fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java

```diff
@@ -26,7 +26,7 @@ public void executeTasks(TaskContext context) {
         long timeout = context.getOptimizerContext().getSessionVariable().getOptimizerExecuteTimeout();
         Stopwatch watch = context.getOptimizerContext().getTraceInfo().getStopwatch();
         while (!tasks.empty()) {
-            if (watch.elapsed(TimeUnit.MILLISECONDS) > timeout) {
+            if (timeout > 0 && watch.elapsed(TimeUnit.MILLISECONDS) > timeout) {
```

Comment:
if we want to valid the user config, we should check it when set, not here.

Reply:
Not a check, here I treat the `timeout =0` as not ignore the timeout